x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawl_detail_logic.test.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  LogicMounter,
+  mockHttpValues,
+  mockFlashMessageHelpers,
+} from '../../../__mocks__/kea_logic';
+import '../../__mocks__/engine_logic.mock';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { CrawlDetailLogic, CrawlDetailValues } from './crawl_detail_logic';
+import { CrawlerStatus, CrawlRequestFromServer } from './types';
+import { crawlRequestServerToClient } from './utils';
+
+const DEFAULT_VALUES: CrawlDetailValues = {
+  dataLoading: true,
+  isFlyoutOpen: false,
+  request: null,
+};
+
+const crawlRequestResponse: CrawlRequestFromServer = {
+  id: '12345',
+  status: CrawlerStatus.Pending,
+  created_at: 'Mon, 31 Aug 2020 17:00:00 +0000',
+  began_at: null,
+  completed_at: null,
+};
+
+const clientCrawlRequest = crawlRequestServerToClient(crawlRequestResponse);
+
+describe('CrawlDetailLogic', () => {
+  const { mount } = new LogicMounter(CrawlDetailLogic);
+  const { http } = mockHttpValues;
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    mount();
+    expect(CrawlDetailLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('reducers', () => {
+    describe('closeFlyout', () => {
+      it('closes the flyout', () => {
+        mount({ isFlyoutOpen: true });
+
+        CrawlDetailLogic.actions.closeFlyout();
+
+        expect(CrawlDetailLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          isFlyoutOpen: false,
+        });
+      });
+    });
+
+    describe('fetchCrawlRequest', () => {
```

Comment:
Logically, I understand why you have a `fetchCrawlRequest` test in actions and listeners. However, I think for someone reading this test, it'd be helpful to have them collapsed to a single test, or at least the same describe.

Reply:
Gotchya.  I know we've traditionally used seperate `actions` and `listeners` blocks but it wasn't clear to me what tests belong in each category.

Reply:
I moved this in a3efc7c

Reply:
True.